File: JPR/src/controllers/analizador/tablaSimbolos/Arbol.ts
import Simbolo from "./Simbolo";
import tablaSimbolos from "./tablaSimbolos";
import Excepcion from "../Excepciones/Excepcion";
import {Instruccion} from "../Abstract/Instruccion";

export default class Arbol {
    private instrucciones: Array<Instruccion>;
    private errores: Array<Excepcion>;
    private consola: String;
    private global: tablaSimbolos;

    constructor(instrucciones: Array<Instruccion>){
        this.instrucciones = instrucciones;
        this.consola = "";
        this.global = new tablaSimbolos();
        this.errores = new Array<Excepcion>();
    }

    public getInstrucciones():Array<Instruccion>{
        return this.instrucciones;
    }

    public setInstrucciones(instrucciones: Array<Instruccion>){
        this.instrucciones = instrucciones;
    }

    public getConsola():String{
        return this.consola;
    }

    public setConsola(consola:String){
        this.consola = consola;
    }

    public updateConsola(update:String){
        this.consola = `${this.consola}${update}\n`;
    }

    public getGlobal():tablaSimbolos{
        return this.global;
    }

    public setGlobal(tabla:tablaSimbolos){
        this.global = tabla;
    }
}

File: JPRJava/JPR/src/TablaSimbolos/Arbol.java
package TablaSimbolos;

import Excepciones.Excepcion;
import Abstract.AST;
import java.util.ArrayList;
import javafx.scene.Group;

/**
 *
 * @author Pavel
 */
public class Arbol {

    private ArrayList<AST> instrucciones;
    private ArrayList<Excepcion> excepciones;
    private String consola;
    private Tabla global;

    public Arbol(ArrayList<AST> instrucciones) {
        this.instrucciones = instrucciones;
        this.excepciones = new ArrayList<>();
    }

    public ArrayList<AST> getInstrucciones() {
        return instrucciones;
    }

    public void setInstrucciones(ArrayList<AST> instrucciones) {
        this.instrucciones = instrucciones;
    }

    public ArrayList<Excepcion> getExcepciones() {
        return excepciones;
    }

    public void setExcepciones(ArrayList<Excepcion> excepciones) {
        this.excepciones = excepciones;
    }
    
    public String getConsola() {
        return consola;
    }

    public void setConsola(String consola) {
        this.consola = consola;
    }
    
    public void updateConsola(String consola) {
        this.consola += consola + "\n";
    }

    public Tabla getGlobal() {
        return global;
    }

    public void setGlobal(Tabla global) {
        this.global = global;
    }

      
}
